CMakeLists.txt

```
@@ -79,6 +79,17 @@ function(FATAL_ERROR_MISSING_ENV)
   endif()
 endfunction()
 
+if(UNIX AND NOT APPLE)
+  execute_process(COMMAND uname -r OUTPUT_VARIABLE UNAME_R OUTPUT_STRIP_TRAILING_WHITESPACE)
+  if(UNAME_R MATCHES ".*[Mm]icrosoft.*" OR UNAME_R MATCHES ".*WSL.*")
+    if(CMAKE_BUILD_TYPE MATCHES "Debug")
+      message(FATAL_ERROR "Building on Windows Subsystem for Linux (WSL) is not supported. For native building you need to use the \"x64 Native Tools Command Prompt for VS 2022\" and download the Mixxx build environment using `${CMAKE_SOURCE_DIR}/tools/windows_buildenv.bat`.")
+    else()
+      message(FATAL_ERROR "Building on Windows Subsystem for Linux (WSL) is not supported. For native building you need to use the \"x64 Native Tools Command Prompt for VS 2022\" and download the Mixxx build environment using `${CMAKE_SOURCE_DIR}/tools/windows_release_buildenv.bat` or `${CMAKE_SOURCE_DIR}/tools/windows_buildenv.bat`(includes Debug).")
+    endif()
```

```suggestion
      message(FATAL_ERROR "Building on Windows Subsystem for Linux (WSL) is not supported. If you want to build Mixxx from command line on Windows, you need to use the \"x64 Native Tools Command Prompt for VS 2022\"!")
```
I suggest to simplify this, the batch file has not much to do with the failure the user did. It makes the message generation complicated and distracts the reader from the core information, which is that he has to use the correct shell.